app/src/main/java/com/architectcoders/arquitectomarvel/data/database/RoomDataSource.kt

```diff
@@ -10,19 +11,44 @@ class RoomDataSource(db: MarvelDatabase) : LocalDataSource {
 
     private val characterDao = db.marvelDao
 
-    override suspend fun getLocalCharacters(): List<CharacterResult> =
+    override suspend fun getLocalCharacters(): List<CharacterResult>? =
         withContext(Dispatchers.IO) {
```

Comment:
I think that wrap with withContext(Dispatcher.IO) is a boilerplate unnecessary.
We can see in google repos like [this](https://github.com/android/sunflower)

Reply:
You are right @joseantoniocampillo ! Check [PlantRepository.kt](https://github.com/android/sunflower/blob/main/app/src/main/java/com/google/samples/apps/sunflower/data/PlantRepository.kt) it says: _Room supports Coroutines and moves the query execution off of the main thread_. Thanks!